### DISCUSSION TITLE
Simulation changes initial control setting

### DISCUSSION THREAD

**Discussion by magoorden** [pyswmm/pyswmm#444] [Q&A]
I experience a strange situation where my desired initial value of an orifice control setting is not the actual value being used by the SWMM simulation.

Below I have a minimized snippit of the code. In this piece of code, I set the initial orifice setting to `0.2` (eventually another piece of code will calculate what this should be). But when the code enters the simulation loop, the actual setting (both target and current) of the orifice is changed to `1`. Hence, the `assert` fails.

In the swmm `.inp` file there is no controls block that might override some settings.

```python
def swmm_control(swmm_inputfile, orifice_id, time_step):
    orifice_settings = []

    with Simulation(swmm_inputfile) as sim:
        orifice = Links(sim)[orifice_id]
        sim.step_advance(time_step)

        orifice.target_setting = 0.2
        orifice_settings.append(orifice.target_setting)

        # Run the simulation.
        for step in sim:
            orifice_settings.append(orifice.target_setting)
            assert orifice_settings[-1] == orifice_settings[-2]  # Sanity check
```

**Comment by karosc** [comment]
This might be related to this bug:

https://github.com/pyswmm/pyswmm/issues/435

Setting an orifice setting outside of the simulation loop currently doesn't work because swmm_start initializes all orifices with a setting of 1.0

**Reply by magoorden** [reply]
It sounds indeed like this bug report.

**Reply by bemcdonnell** [reply]
I am going to write a couple of tests using the `sim.before_start()` and `sim.initial_conditions()` to also confirm that behavior.  I remember there was a pestering limitation with how the beginning of the simulation was handled with SWMM between `swmm_start` `swmm_step`.  If I remember correctly, we have always just accepted that the first timestep was burned
